watertap/flowsheets/full_water_resource_recovery_facility/BSM2.py

```diff
@@ -385,20 +401,838 @@ def set_operating_conditions(m, reactor_volume_equalities=False):
+        },
+    )
+    # Poorly scaled Jacobians
+    sb.set_variable_scaling_factor(m.fs.RADM.liquid_phase.heat[0], 1e-3, overwrite=True)
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.enthalpy_transfer[0], 1e-2, overwrite=True
+    )
+    sb.set_variable_scaling_factor(m.fs.RADM.liquid_phase.reactions[0].S_H, 1e7)
+
+    for c in m.fs.props_vap.solute_set:
+        sb.set_variable_scaling_factor(
+            m.fs.RADM.vapor_phase[0].conc_mass_comp[c], 1e2, overwrite=True
+        )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.vapor_phase[0].conc_mass_comp["S_h2"], 1e3, overwrite=True
+    )
+
+    for c in m.fs.props_vap.component_list:
+        sb.set_variable_scaling_factor(
+            m.fs.RADM.vapor_phase[0].pressure_sat[c], 1e-3, overwrite=True
+        )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.properties_in[0].conc_mass_comp["X_aa"],
+        1,
+        overwrite=True,
+    )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.properties_in[0].conc_mass_comp["X_h2"],
+        1,
+        overwrite=True,
+    )
+
+    sb.set_variable_scaling_factor(m.fs.RADM.KH_h2[0], 1e4)
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.KH_ch4[0], 1e3
+    )  # NOTE: This drastically reduced the number of iterations
+    for c in m.fs.props_ADM1.solute_set:
+        sb.set_variable_scaling_factor(
+            m.fs.RADM.liquid_phase.mass_transfer_term[0, "Liq", c],
+            1e10,
+            overwrite=True,
+        )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.mass_transfer_term[0, "Liq", "H2O"],
+        1e4,
+        overwrite=True,
+    )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.mass_transfer_term[0, "Liq", "S_h2"],
+        1e2,
+        overwrite=True,
+    )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.mass_transfer_term[0, "Liq", "S_ch4"],
+        1e2,
+        overwrite=True,
+    )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.mass_transfer_term[0, "Liq", "S_cat"],
+        1e9,
+        overwrite=True,
+    )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.mass_transfer_term[0, "Liq", "S_an"],
+        1e9,
+        overwrite=True,
+    )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.mass_transfer_term[0, "Liq", "S_IC"],
+        1e3,
+        overwrite=True,
+    )
+
+    for rxn in m.fs.ADM1_rxn_props.rate_reaction_idx:
+        sb.set_variable_scaling_factor(
+            m.fs.RADM.liquid_phase.reactions[0].reaction_rate[rxn],
+            1e7,
+            overwrite=True,
+        )
+
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.reactions[0].reaction_rate["R1"],
+        1e5,
+        overwrite=True,
+    )
+    sb.set_variable_scaling_factor(
+        m.fs.RADM.liquid_phase.reactions[0].reaction_rate["R11"],
+        1e5,
+        overwrite=True,
+    )
```

Comment:
I'm unclear on why we would us `sb`, a separate ScalerBase instance, to apply manual scaling factors to a unit as opposed to using the instance of ADScaler to set all of the variable scale factors manually.

Reply:
For application specific scaling factors, you should just set the scaling factors directly without using the base class. You should generally not be writing case-specific scaling classes.

Scalers are meant to help automate the common tasks, not handle the case-specific scaling. For those cases, use things like `set_scaling_factor` from the utilities directly (this should mostly only involve scaling variables, as constraints can generally be automated).